### PR TITLE
Remove mitie support

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- mitie is to cumbersome to maintain over time and the benefit of having it is quite small as it can not deal with larger datasets - we will replace it with a different backend alternative

**TODO**
- lookup parameters for mitie intent classifier to mimik its classification performance (which will probably also mimik its long training times :D )

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: docs/entities.rst [hunk 1]
```diff
@@ -45,7 +44,7 @@ Custom, Domain-specific entities
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 In the introductory tutorial we build a restaurant bot, and create custom entities for location and cuisine.
-The best components for training these domain-specific entity recognisers are the ``ner_mitie`` and ``ner_crf`` components. 
+The best components for training these domain-specific entity recognisers is the ``ner_crf`` component.
 It is recommended that you experiment with both of these to see what works best for your data set.
```
Comment (akelad) [inline docs/entities.rst]: no longer relevant

Comment (tmbo) [reply docs/entities.rst]: why not? makes clear that you should use it instead of duckling for your domain specific entities

Comment (akelad) [reply docs/entities.rst]: I'm talking about this sentence "It is recommended that you experiment with both of these to see what works best for your data set."  😄

Comment (tmbo) [reply docs/entities.rst]: ah yes, sure :D

**Pull request merged**